src/snapred/backend/recipe/ReductionGroupProcessingRecipe.py

```diff
@@ -0,0 +1,108 @@
+from typing import Any, Dict, List, Tuple
+
+from snapred.backend.log.logger import snapredLogger
+from snapred.backend.recipe.algorithm.MantidSnapper import MantidSnapper
+from snapred.backend.recipe.algorithm.Utensils import Utensils
+from snapred.meta.decorators.Singleton import Singleton
+
+logger = snapredLogger.getLogger(__name__)
+
+Pallet = Tuple[Dict[str, str]]
+
+
+@Singleton
+class ReductionGroupProcessingRecipe:
+    def __init__(self, utensils: Utensils = None):
+        """
+        Sets up the recipe with the necessary utensils.
+        """
+        # NOTE: workaround, we just add an empty host algorithm.
+        if utensils is None:
+            utensils = Utensils()
+            utensils.PyInit()
+        self.mantidSnapper = utensils.mantidSnapper
```

Comment:
I must have missed this, is this some way of making a fake algorithm to be a parent to a `MantidSnapper` object?